app/assets/javascripts/components/timeline/TrainingModules/TrainingModulesViewMode.jsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import createReactClass from 'create-react-class';
+import PropTypes from 'prop-types';
+import TrainingModules from '../TrainingModules/TrainingModules';
+import {
+    DISCUSSION_KIND, EXERCISE_KIND
+  } from '../../../constants';
+
+const TrainingModulesViewMode = createReactClass({
+    displayName: 'TrainingModulesViewMode',
+
+    propTypes: {
+        block: PropTypes.object,
+        all_training_modules: PropTypes.array,
+        trainingLibrarySlug: PropTypes.string.isRequired,
+        editable: PropTypes.bool,
+      },
+
+      passedUpdateBlock(selectedIds) {
```

Comment:
Is this update code necessary in view mode? If so, it can probably be passed from the parent component instead of duplicated here. But I suspect it can be removed altogether, since the view-mode components shouldn't need to fire any update actions.

Reply:
You are right, this is unnecessary.